.github/workflows/consensus-tla.yml

```diff
@@ -0,0 +1,144 @@
+name: "TLA+ Spec Verification - Consensus"
+
+on:
+  schedule:
+    - cron: "0 0 * * 0"
+  pull_request:
+    # TO DO update to tla change only / weekly
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  actions: read
+  contents: read
+  security-events: write
+
+jobs:
+  model-checking-consensus:
+    name: Model Checking - Consensus
+    runs-on: [self-hosted, 1ES.Pool=gha-virtual-ccf-sub]
+    container:
+      image: ghcr.io/microsoft/ccf/ci/default:build-26-06-2024
+
+    steps:
+      - uses: actions/checkout@v4
+      - run: |
+          sudo apt update
+          sudo apt install -y default-jre
+          python3 ./tla/install_deps.py
+
+      - name: MCccfraft.cfg
+        run: |
+          cd tla/
+          ./tlc.sh -workers auto consensus/MCccfraft.tla -dumpTrace tla MCccfraft.trace.tla -dumpTrace json MCccfraft.json
+
+      - name: MCccfraftAtomicReconfig.cfg
+        run: |
+          cd tla/
+          JVM_OPTIONS=-Dtlc2.TLCGlobals.coverage=1 ./tlc.sh -workers auto -config consensus/MCccfraftAtomicReconfig.cfg consensus/MCccfraft.tla -dumpTrace tla MCccfraftAtomicReconfig.trace.tla -dumpTrace json MCccfraftAtomicReconfig.json
+
+      - name: MCccfraftWithReconfig.cfg
+        run: |
+          cd tla/
+          JVM_OPTIONS=-Dtlc2.TLCGlobals.coverage=1 ./tlc.sh -workers auto -config consensus/MCccfraftWithReconfig.cfg consensus/MCccfraft.tla -dumpTrace tla MCccfraftWithReconfig.trace.tla -dumpTrace json MCccfraftWithReconfig.json
+
+      - name: Coverage
+        run: |
+          cd tla/
+          python3 -mvenv env
+          source env/bin/activate
+          pip install -r requirements.txt
+          ls -tr MCccfraftAtomicReconfig_coverage.json | xargs cat | python3 actions.py MCccfraftAtomicReconfig.html
+          rm MCccfraftAtomicReconfig_coverage.json
+          ls -tr MCccfraftWithReconfig_coverage.json | xargs cat | python3 actions.py MCccfraftWithReconfig.html
+          rm MCccfraftWithReconfig_coverage.json
```

Comment:
I'd suggest archiving`*_coverage.json` as workflow artifact instead of deleting the files.

Reply:
I see you added that `rm` in #6092 

Are you sure we want to get them back and `upload_artifact` them?

Reply:
I don't remember adding those lines.  @achamayou, was it you who initially added that block?

Reply:
Ok, I just checked the blame diff, you didn't add `rm`, you were doing renaming only

https://github.com/microsoft/CCF/pull/6092/commits/43ae605c633c43267c42c784eb08ef6f743a40ed

Reply:
The coverage files were very large, took a long time to upload, and also caused our other more useful artefacts like node logs to be deleted sooner as we run out of quota. It's possible the quota has changed now that we're on GH, but that needs to be checked first.

Reply:
> The coverage files were very large, took a long time to upload, and also caused our other more useful artefacts like node logs to be deleted sooner as we run out of quota. It's possible the quota has changed now that we're on GH, but that needs to be checked first.

I would not have suggested archiving the coverage results if I had known this.  Perhaps we should add a comment explaining why the coverage data is not archived.

Reply:
It was reasonable to try - we did! If we had a CI system where upload is super faster and storage unlimited, we would store everything. I seem to remember that even just enabling coverage in TLC made the model checking much slower though: https://github.com/microsoft/CCF/pull/5702